generated/niscope/niscope/_grpc_stub_interpreter.py

```diff
@@ -0,0 +1,425 @@
+                        error_message += f'\nError status: {value}'
+            grpc_error = rpc_error.code()
+
+        if grpc_error == grpc.StatusCode.UNAVAILABLE:
+            error_message = 'Failed to connect to server'
+        elif grpc_error == grpc.StatusCode.UNIMPLEMENTED:
+            error_message = (
+                'This operation is not supported by the NI gRPC Device Server being used. Upgrade NI gRPC Device Server.'
+            )
+        elif grpc_error == grpc.StatusCode.NOT_FOUND:
+            raise errors.DriverTooOldError()
+
+        if grpc_error == grpc.StatusCode.INVALID_ARGUMENT:
+            raise ValueError(error_message)
+        elif error_code is None:
+            raise errors.Error(error_message)
+        elif error_code < 0:
+            raise errors.DriverError(error_code, error_message)
+        elif error_code > 0:
+            if not error_message:
+                try:
+                    error_message = self.error_message(error_code)
+                except errors.Error:
+                    error_message = 'Failed to retrieve error description.'
+            warnings.warn(errors.DriverWarning(error_code, error_message))
+        return response
+
+    def abort(self):  # noqa: N802
+        self._invoke(
+            self._client.Abort,
+            grpc_types.AbortRequest(vi=self._vi),
+        )
+
+    def acquisition_status(self):  # noqa: N802
+        response = self._invoke(
+            self._client.AcquisitionStatus,
+            grpc_types.AcquisitionStatusRequest(vi=self._vi),
+        )
+        return enums.AcquisitionStatus(response.acquisition_status_raw)
+
+    def actual_meas_wfm_size(self, array_meas_function):  # noqa: N802
+        response = self._invoke(
+            self._client.ActualMeasWfmSize,
+            grpc_types.ActualMeasWfmSizeRequest(vi=self._vi, array_meas_function_raw=array_meas_function.value),
+        )
+        return response.meas_waveform_size
+
+    def actual_num_wfms(self, channel_list):  # noqa: N802
+        response = self._invoke(
+            self._client.ActualNumWfms,
+            grpc_types.ActualNumWfmsRequest(vi=self._vi, channel_list=channel_list),
+        )
+        return response.num_wfms
+
+    def add_waveform_processing(self, channel_list, meas_function):  # noqa: N802
+        self._invoke(
+            self._client.AddWaveformProcessing,
+            grpc_types.AddWaveformProcessingRequest(vi=self._vi, channel_list=channel_list, meas_function_raw=meas_function.value),
+        )
+
+    def auto_setup(self):  # noqa: N802
+        self._invoke(
+            self._client.AutoSetup,
+            grpc_types.AutoSetupRequest(vi=self._vi),
+        )
+
+    def cal_fetch_date(self, which_one):  # noqa: N802
+        raise NotImplementedError('cal_fetch_date is not supported over gRPC')
+
+    def cal_fetch_temperature(self, which_one):  # noqa: N802
+        raise NotImplementedError('cal_fetch_temperature is not supported over gRPC')
```

Comment:
It doesn't matter for this PR, because if it's not implemented yet, this error case makes sense. But do you know why some of these APIs aren't there for gRPC? Are these new (such that we should go back and add them to gRPC) or old (obsolete, and we shouldn't worry about it)?

Reply:
We have not added any entrypoints to the NI-SCOPE API this year, as far as I know.

Reply:
I don't know of any reason why either of these cal_ methods shouldn't be supported by gRPC.

Reply:
Perhaps it comes from a misconception that an external cal session is needed to call them?

Reply:
Ah, I found it in an old [deleted changes.md](https://github.com/ni/grpc-device/pull/691/files#diff-a994c5c8a10fa7b839bf24e2c7468566970a29c032be3dd4a5f16c73e640ac4a) we had originally in grpc-device to list changes from the nimi-python metadata. Looks like for these we didn't add them because they weren't in niScope.h but instead in niScopeCal.h.

I don't know if there was any other reason except that. Seems like something we could eventually support in grpc-device.

Reply:
Yeah, most of the functions in that header file probably require an external cal session, but these 2 can also be called from a regular session.

Reply:
Ah, I see. Yeah, sounds reasonable to try to get those added to grpc-device then. Added a separate task to look into and try to support these two methods on the grpc-device side so here we can pass through instead of raising a NotImplementedError: [AB#2210201](https://dev.azure.com/ni/DevCentral/_workitems/edit/2210201)

Reply:
@danestull we did have these NotImplementedError tests, but they seem to have gotten lost (probably during the merge of main). Given that the work to add support for these is alreeady underway, I'll allow the omission of these NotImplemented tests for grpc for this PR, so long as there's a TODO above them in the TestLibrary class. I've already requested that TODO in a separate thread, so I'll mark this thread as resolved.